app/models/allocation_version.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+class AllocationVersion < ApplicationRecord
+  has_paper_trail
+
+  attr_accessor :responsibility
+
+  ALLOCATE_PRIMARY_POM = 0
+  REALLOCATE_PRIMARY_POM = 1
+  ALLOCATE_SECONDARY_POM = 2
+  REALLOCATE_SECONDARY_POM = 3
+  DEALLOCATE_PRIMARY_POM = 4
+  DEALLOCATE_SECONDARY_POM = 5
+
+  USER = 0
+  OFFENDER_MOVEMENT = 1
+
+  enum event: {
+    allocate_primary_pom: ALLOCATE_PRIMARY_POM,
+    reallocate_primary_pom: REALLOCATE_PRIMARY_POM,
+    allocate_secondary_pom: ALLOCATE_SECONDARY_POM,
+    reallocate_seondary_pom: REALLOCATE_SECONDARY_POM,
+    deallocate_primary_pom: DEALLOCATE_PRIMARY_POM,
+    deallocate_secondary_pom: DEALLOCATE_SECONDARY_POM
+  }
+
+  enum event_trigger: {
+    user: USER,
+    offender_movement: OFFENDER_MOVEMENT
+  }
+
+  scope :allocations, lambda { |nomis_offender_ids|
+    where(nomis_offender_id: nomis_offender_ids)
+  }
+  scope :all_primary_pom_allocations, lambda { |nomis_staff_id|
```

Comment:
This seems to be all allocations for a primary POM, where the event was an allocate or reallocate.
If the last event on an allocation was neither of these (for instance assigning a coworker) does that mean it won't show up as an allocation for the POM?

Reply:
Yes currently that is true and would need to be modified for co-worker once this ticket is worked on. I'll add a todo